cajita/src/Cajita_HypreStructuredSolver.hpp

```diff
@@ -32,22 +33,78 @@
 #include <numeric>
 #include <sstream>
 #include <string>
+#include <type_traits>
 #include <vector>
 
 namespace Cajita
 {
+//---------------------------------------------------------------------------//
+// Hypre memory space selection. Don't compile if HYPRE wasn't configured to
+// use the device.
+// ---------------------------------------------------------------------------//
+
+//! Hypre device compatibility check.
+template <class MemorySpace>
+struct HypreIsCompatibleWithMemorySpace : std::false_type
+{
+};
+
+// FIXME: This is currently written in this structure because HYPRE only has
+// compile-time switches for backends and hence only one can be used at a
+// time. Once they have a run-time switch we can use that instead.
+#if defined HYPRE_USING_CUDA
```

Comment:
Apparently it's `HYPRE_WITH_CUDA`

Reply:
`HYPRE_config.h` has `HYPRE_USING_CUDA` and `HYPRE_USING_HIP`